### ISSUE TITLE
`torch.pow` and `torch.float_power` have different gradient with `bfloat16` input tensor 

### ISSUE THREAD

**Issue by upupyy** [pytorch/pytorch#76645]
### 🐛 Describe the bug

`torch.pow` and `torch.float_power` have different gradient with `bfloat16` input tensor. For example, the gradients of `exponent` are not equal.

```python
import torch
input_tensor = torch.rand([1], dtype=torch.bfloat16)
exponent_tensor = torch.rand([5, 5, 5], dtype=torch.float64)

input = input_tensor.clone().detach().requires_grad_()
exponent = exponent_tensor.clone().detach().requires_grad_()
res_1 = torch.float_power(input, exponent)
res_1.sum().backward()
grad_1 = [input.grad, exponent.grad]

input = input_tensor.clone().detach().requires_grad_()
exponent = exponent_tensor.clone().detach().requires_grad_()
res_2 = torch.pow(input, exponent)
res_2.sum().backward()
grad_2 = [input.grad, exponent.grad]

print(torch.allclose(res_1, res_2))
for i in range(len(grad_1)):
    print(f"Grad {i}:", torch.allclose(grad_1[i], grad_2[i]))
# True
# Grad 0: True
# Grad 1: False
```
However, when the input has `float32` or `float64` dtype, all of the gradients are equal.

### Versions

pytorch: 1.11.0

**Comment by mruberry** [comment]
Isn't this expected because float_power computes in double but pow computes in bfloat16?

**Comment by upupyy** [comment]
> Isn't this expected because float_power computes in double but pow computes in bfloat16?

I think so. Thank you for your explanation! Let me close this issue.
